### ISSUE TITLE
"Unread messages" causes screen to jump when appearing/disappearing

### ISSUE THREAD

**Issue by WesleyAC** [zulip/zulip-mobile#4526]
The blue "Unread messages" bar at the top of the screen causes the message view window to noticeably jump when it appears/disappears. Instead, it should not affect the rendering of the message list at all, but smoothly appear on top of it, and then disappear again once the messages are read.

I think the fix for this will look like setting `position: absolute` on `UnreadNotice`, but I suspect it'll take a little bit of finagling to get it just right.

**Comment by chrisbobbe** [comment]
Yeah; that jumping would be good to fix. In particular, I think the banner-exiting works like this (as far as I remember; I can double-check when I'm next at my regular dev setup):

1. The banner begins its scale animation (which we're not fond of), decreasing in size while remaining centered in the rectangle that's laid out for it. The top of the WebView doesn't move; it remains aligned with where the bottom of the banner was before its exit animation.
2. The banner progresses through its exiting animation. The top of the WebView still doesn't move.
3. The banner finishes its exiting animation. The top of the WebView "jumps" up, so the WebView fills the space that was previously occupied by the banner.

(ISTR the banner's entrance works like that but in reverse; the top of the WebView "jumps" down to make room for the banner, then the banner animates in.)

> I think the fix for this will look like setting `position: absolute` on `UnreadNotice`, but I suspect it'll take a little bit of finagling to get it just right.

One property of the current layout that would be good to keep (Greg reminded me of how this works in https://github.com/zulip/zulip-mobile/issues/4209#issuecomment-665444705, which is why it comes to mind) is that it's impossible for the unread banner to obscure anything in the message list. That's currently done by stacking the unread banner and the WebView vertically in the layout, in the y-direction. I think if we used absolute positioning and a Z-index to overlay the banner on top of the WebView, we'd risk the banner obscuring some of the message list's content.

I'd be very interested to know if we could fix the jumping, and keep the non-obscuring behavior, by ditching the current animation and (if we wanted there to be an animation at all) using React Native's [`LayoutAnimation` API](https://reactnative.dev/docs/layoutanimation) to have the banner's position and the WebView's height slide _together_, synchronized. I've recently had good results with that API in the lightbox, in #4442 (see in particular 3fa7a7f10). You can basically tell it to "please schedule the next layout change (whatever that is) so that the elements move smoothly to their new positions."

**Comment by gnprice** [comment]
I agree, this is a visible annoyance that'd be good to fix.

I think it may be tricky to work out a behavior here that's just right. Some constraints that would be good to satisfy are:
* Things shouldn't jump around -- that's this issue.
* As Chris says, the banner shouldn't obscure any of the message list so that you can't see it. In particular, if you scroll to the top of the history we want you to be able to see what's at the top (a date stamp, as it happens). Currently we successfully accomplish this.
* When the banner appears and disappears, it shouldn't push the messages around. I think currently we accomplish this on net, in the end -- there's this jerking around, but the messages wind up at the same position after the jerking as they were before.
  * This is a bit tricky to do. See #4043 for some background, and #3301.
  * I think if we replaced the current half-animated-half-abrupt way the banner disappears with a smooth sliding animation (with e.g. `LayoutAnimation` as @chrisbobbe mentioned), so that the WebView's height got smoothly animated, that would disrupt our current implementation of how we try to maintain this constraint. So that'd be something to watch out for and try to deal with.

Another direction would be to try to replace the "N unread messages" banner with some other UI that doesn't pose these layout and animation challenges. I don't have concrete ideas at the moment for what that would be, but we could try to imagine one.

**Comment by WesleyAC** [comment]
Some notes on CZO in [#mobile-team > #M4526, unread bar jank](https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/.23M4526.2C.20unread.20bar.20jank).

As it turns out, this issue is kinda Big and Complicated, I'll split up some of my findings into smaller issues.

**Comment by Jumeb** [comment]
> I agree, this is a visible annoyance that'd be good to fix.
> 
> I think it may be tricky to work out a behavior here that's just right. Some constraints that would be good to satisfy are:
> 
> * Things shouldn't jump around -- that's this issue.
> * As Chris says, the banner shouldn't obscure any of the message list so that you can't see it. In particular, if you scroll to the top of the history we want you to be able to see what's at the top (a date stamp, as it happens). Currently we successfully accomplish this.
> * When the banner appears and disappears, it shouldn't push the messages around. I think currently we accomplish this on net, in the end -- there's this jerking around, but the messages wind up at the same position after the jerking as they were before.
>   
>   * This is a bit tricky to do. See [Fix scroll-on-resize to avoid jumping from bottom #4043](https://github.com/zulip/zulip-mobile/pull/4043) for some background, and [Message list scroll position is often slightly messed up #3301](https://github.com/zulip/zulip-mobile/issues/3301).
>   * I think if we replaced the current half-animated-half-abrupt way the banner disappears with a smooth sliding animation (with e.g. `LayoutAnimation` as @chrisbobbe mentioned), so that the WebView's height got smoothly animated, that would disrupt our current implementation of how we try to maintain this constraint. So that'd be something to watch out for and try to deal with.
> 
> Another direction would be to try to replace the "N unread messages" banner with some other UI that doesn't pose these layout and animation challenges. I don't have concrete ideas at the moment for what that would be, but we could try to imagine one.

I think the problem comes from switching the `display` property from **flex** to **none**. I discussed this with @alya [here](https://github.com/zulip/zulip-mobile/issues/4930#issuecomment-957750324)
